tests/test_summarizer.py

```
@@ -0,0 +1,364 @@
+
+                with pytest.raises(ImportError):
+                    summarizer_module.DocumentSummarizer()
+
+    @patch("src.chunker.summarizer.INSTRUCTOR_AVAILABLE", True)
+    @patch("src.chunker.summarizer.instructor")
+    def test_init_with_custom_provider(self, mock_instructor):
+        """Should accept custom provider."""
+        mock_instructor.from_provider.return_value = MagicMock()
+
+        from src.chunker.summarizer import DocumentSummarizer
+        _summarizer = DocumentSummarizer(provider="anthropic/claude-opus-4-20250514")  # noqa: F841
```

Ruff (0.14.10)

Unused `noqa` directive (unused: `F841`)

Remove unused `noqa` directive

(RUF100)